test/e2e/suites/e2e/e2e_test.go

```diff
@@ -605,17 +614,14 @@ var _ = Describe("e2e tests [PR-Blocking]", func() {
 	})
 })
 
-func createCluster(ctx context.Context, configCluster clusterctl.ConfigClusterInput) []*clusterv1.MachineDeployment {
-	result := &clusterctl.ApplyClusterTemplateAndWaitResult{}
+func createCluster(ctx context.Context, configCluster clusterctl.ConfigClusterInput, result *clusterctl.ApplyClusterTemplateAndWaitResult) {
```

Comment:
The name of the cluster is needed when collecting logs. I had to refactor this a bit to make the name known in the `AfterEach`. The cluster object is part of the result here so we can get the name that way. This is also mimicking what CAPI is doing (although they use `ApplyClusterTemplateAndWaitResult` directly).